TaskLayer/SearchTask/PostSearchAnalysisTask.cs

```diff
@@ -1017,59 +1023,220 @@ private void WriteVariantResults()
             }
             foreach (var peptide in confidentVariantPeps)
             {
-                if (peptide.FdrInfo.QValue <= 0.01 && peptide.FdrInfo.QValueNotch <= 0.01 && peptide.IsDecoy == false)
+                if (peptide.FdrInfo.QValue <= 0.01 && peptide.FdrInfo.QValueNotch <= 0.01 && !peptide.IsDecoy && !peptide.IsContaminant)
                 {
                     var variantPWSM = peptide.BestMatchingPeptides.FirstOrDefault();//TODO: expand to all peptide options not just the first
                     var variants = variantPWSM.Peptide.Protein.AppliedSequenceVariations;
                     var culture = CultureInfo.CurrentCulture;
+                    // these bools allow for us to accurrately count the number of peptides that have at least one variants of a given type. 
+                    // they will prevent double counting if a variant type is found more than once in a given peptide (most typically missense, but all will be covered)
+                    bool SNVmissenseIdentified = false;
+                    bool MNVmissenseIdentified = false;
+                    bool insertionIdentified = false;
+                    bool deletionIdentified = false;
+                    bool frameshiftIdentified = false;
+                    bool stopGainIdentified = false;
+                    bool stopLossIdentifed = false;
+
                     foreach (var variant in variants)
                     {                       
                         if (variantPWSM.Peptide.IntersectsAndIdentifiesVariation(variant).identifies == true)
```

Comment:
This unit test does not read in a fake database made by spritz. Instead a protein database is generated internally with a fake protein and fake variant. Becasue of this, the variant description does not follow the same format as what is produced by spritz and doesn't contain all of the information such as the variant type. This is why the SNVmissenseCount is not incremented. This is not an error in the code, and the string that is passed in as the varaint description could be altered to contain "missense_variant"  if deemed necessary.

Reply:
should we make it so that a defined variant peptides has defined/required fields that will provide the necessary data for correct analysis downstream?

Reply:
I do not think that is necessary. The SequenceVariation object already has defined/ required fields. The variant description would normally be extracted from the proteogenomic database and would always contain the variant type. But since a fake variant was made, a fake variant description was made that did not have the variant type in it which would only occur in the case of a unit test.